compiler/z/codegen/ConstantDataSnippet.cpp

```diff
@@ -195,6 +195,13 @@ TR::S390ConstantDataSnippet::addMetaDataForCodeAddress(uint8_t *cursor)
             }
          break;
 
+      case TR_CatchBlockCounter:
+         {
+         TR::Relocation *relocation = new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor, NULL, TR_CatchBlockCounter, cg());
```

Comment:
This doesn't seem to follow the same patters as in the other cases above.

Reply:
The value of the second argument is one of the "target pointers" of the relocation, which is just some arbitrary pointer-sized data that a relocation can hold. The `TR_CatchBlockCounter` relocation doesn't need to use it, since the counter is at a fixed location in the persistent method info at relocation time.

I'm not entirely sure why the `TR_AbsoluteMethodAddress` or `TR_BodyInfoAddress` save the value of the address at the cursor; the latter doesn't need it for the same reason that `TR_CatchBlockCounter` doesn't, and I think the former reads off that value from the reloLocation during relocation anyway. Maybe the `TR_AbsoluteMethodAddress` wants to make sure that relocations for distinct addresses don't get merged together, but I haven't looked into why it would want to do that.

Reply:
@cjjdespres explanation makes sense. Looking around the codebase, I don't see any issues related to this so we should be okay. Tagging @r30shah here in case he's aware of a particular reason why we would need the cursor.

Reply:
@cjjdespres  is right, Looking into the code around the `TR_AbsoluteMethodAddress` , I do not think it would need the content of the what is stored at cursor. Also in other places where we would need `TR_AbsoluteMethodAddress` , it does not pass in the content from the cursor. 

[1]. https://github.com/eclipse-openj9/openj9/blob/fdbd742218639e847626d0b3b4a0a20a74c9407d/runtime/compiler/z/codegen/S390J9CallSnippet.cpp#L747-L750